desc/magnetic_fields/_core.py

```diff
@@ -537,6 +613,28 @@ def save_mgrid(
         )
         bz_001[:] = B_Z
 
+        if A_R:
```

Comment:
does this work? I thought using an array as a bool caused issues?

Reply:
looking at it I feel like it should not work and this part of the code is actually untested rn too. I will see if we can add a test and then fix this too